data_juicer/ops/base_op.py

```diff
@@ -319,15 +319,11 @@ def __init__(self, *args, **kwargs):
                 skip_op_error=self.skip_op_error,
                 op_name=self._name)
 
-    # set the process method is not allowed to be overridden
+    @classmethod
     def __init_subclass__(cls, **kwargs):
-        not_allowed_list = ['process']
-        for method_name in not_allowed_list:
-            if method_name in cls.__dict__:
-                raise TypeError(
-                    f'Method {method_name} cannot be overridden by subclass '
-                    f'{cls.__name__}. Please implement {method_name}_single '
-                    f'or {method_name}_batched.')
+        """Register the operator to the registry."""
+        super().__init_subclass__(**kwargs)
+        OPERATORS.register_module(cls.__name__.lower())(cls)
```

Comment:
Difference between this implementation and adding wrapper before class definition?   Does this new modification introduce double registration?